extensions/notebook/src/common/utils.ts

```diff
@@ -144,22 +144,31 @@ export function getOSPlatformId(): string {
  * @returns 1 if the first version is greater, -1 if it's less, and 0 otherwise.
  */
 export function comparePackageVersions(first: string, second: string): number {
-	let firstVersion = first.split('.').map(numStr => Number.parseInt(numStr));
-	let secondVersion = second.split('.').map(numStr => Number.parseInt(numStr));
+	let firstVersion = first.split('.');
```

Comment:
Why not use a package like semver instead? We do that in most other places we need to compare versions.

Reply:
Semver only takes exactly 3 version numbers, from what I remember. I remember using it way back when we first added this, but there were some package cases that semver errored on.

Reply:
Looks like semver.coerce() can convert a version string into a valid semver format, but then you lose information. From the semver usage docs:
`semver.valid(semver.coerce('v2')) // '2.0.0'`

`semver.valid(semver.coerce('42.6.7.9.3-alpha')) // '42.6.7'`

Reply:
https://www.npmjs.com/package/semver#usage